keps/sig-node/3759-sidecar-containers/README.md

```diff
@@ -0,0 +1,1246 @@
+could be used to perform computation outside the allowed limit.
+
+### Scenario 7. Risk of porting existing sidecars to the new mechanism naively
+
+There is risk associated with people moving sidecars as implemented today to use
+the new pattern. We didn’t receive any feedback on potential downsides. One
+scenarios that may be affected is if sidecars decided to terminate itself and
+kubelet keeps trying to restart it as one of the main containers are still being
+terminated. Based on current patterns for sidecar containers, this is not likely
+the problem. 
+
+Another potential problem may be that sidecars will wait for some condition to
+mark itself “started” that cannot be met with the new pattern. For example, wait
+for other containers to fully start. As sidecar will not succeed startup probes,
+other Init containers will not run and Pod will be stuck on initialization. For
+example, Knative's sidecar does aggressive probing of the user's container to
+ensure they're ready prior to marking the sidecar ready itself (not for startup
+probes though). This prevents the Pod from being included K8s ready endpoints.
+However this problem will be discovered right away during the testing stage and
+changes will need to be made to sidecar implementation.
+
+Switching to the new sidecars approach will slow down Pod start for Istio. Istio
+today is not blocking other containers to start during its initialization. With
+the switch to the new model, the separation of Initialization stage and main
+containers running stage will be more explicit and many implementations will
+likely wait for sidecar initialization, effectively slowing down Pod startup
+comparing to current approach. This can be eliminated by slight redesign of a
+sidecars.
+
+
+## Design Details
+
+### Backward compatibility
+
+The new field means that any Pod that is not using this field will behave the
+same way as before.
+
+The new field will only work with the proper control plane and kubelet. Upgrade
+and downgrade scenarios will be covered in the further sections.
+
+Outside of Kubernetes-controlled code, there might be third party controllers or
+existing containers relying on the current behavior of Init containers.
+Behaviors they can rely on:
+
+- Assuming nothing is running in the Pod when Init container starts. For
+  instance taking PID of the process assuming there are no sidecars.
+- Incorrectly calculating the resource usage of a Pod not taking the new type of
+  containers into account (e.g. some grafana dashboards may need modification)
+- Stripping the restartPolicy:Always from Init container as an unknown field
+  rendering Pod unfunctional as it will not pass the Init stage after this
+- OPA rules may reject the Pod with new unknown fields because of failure to
+  parse the new field
+
+These potential incompatibilities will be documented.
+
+### Resources calculation for scheduling and pod admission
```

Comment:
Might be worth making it more explicit that we would have to enhance the [NodeResourcesFit scheduler plugin](https://github.com/kubernetes/kubernetes/blob/release-1.26/pkg/scheduler/framework/plugins/noderesources/fit.go#L160-L176) and take sidecar container resource request into consideration?

Reply:
added